### DISCUSSION TITLE
Error when create item with bw : Cannot encrypt cipher for organization. No key.

### DISCUSSION THREAD

**Discussion by Flo-Apitech** [dani-garcia/vaultwarden#4374] [Q&A] [answered]
Hi,

I have an error when I create a new item with bw.
Edit item, it's ok.

`
echo $ITEM_JSON | $PATH_BINARY_BITWARDEN encode | $PATH_BINARY_BITWARDEN create item
`
Result : 
`
/snapshot/clients/apps/cli/build/bw.js:24003
                    throw new Error("Cannot encrypt cipher for organization. No key.");
                          ^

Error: Cannot encrypt cipher for organization. No key.
    at CipherService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:24003:27)
    at Generator.next (<anonymous>)
    at fulfilled (/snapshot/clients/apps/cli/build/bw.js:23871:58)
`

I use Vaultwarden on my own docker server. 
Image docker : vaultwarden/server:latest 
Version is 2023.7.1

BW version : 2023.4.0
I have update to lastest : 2024.2.0

New error : 
`
/snapshot/clients/apps/cli/build/bw.js:27111
                        throw new Error("Cannot encrypt cipher for organization. No key.");
                              ^

Error: Cannot encrypt cipher for organization. No key.
    at CipherService.<anonymous> (/snapshot/clients/apps/cli/build/bw.js:27111:31)
    at Generator.next (<anonymous>)
    at fulfilled (/snapshot/clients/apps/cli/build/bw.js:27000:58)

Node.js v18.5.0
`

Have you an idea ?

Thanks
Flo

**Comment by Flo-Apitech** [comment]
Hi,

Any idea ?

**Reply by BlackDex** [reply, marked as answer]
Not really.
This project does not maintain the clients, and this is a client error.

Only thing i can think of is that you were not logged-in or didn't unlocked the vault, or it may need a sync first.

**Reply by Flo-Apitech** [reply]
Ok thank for your anwser. 
My vault is unlock, I can update an existing entry. I can test to force sync before.

**Reply by BlackDex** [reply]
Also, you know that the cli has a `serve` option, which let you use cli in a REST API way.
https://bitwarden.com/help/cli/#serve

**Reply by BlackDex** [reply]
Have you tried without all the piping btw?

**Reply by Flo-Apitech** [reply]
Hi,
Thank for option "serve", I haven’t tested it yet.

For the piping, I don't understand, you reference "server" option ?
I have no problem when updating existing items with this command : 
`
echo $ITEM_JSON | $PATH_BINARY_BITWARDEN encode | $PATH_BINARY_BITWARDEN edit item $ITEM_ID
`
in the same script, before I get the ITEM_ID with : 
`
ITEM_ID=`$PATH_BINARY_BITWARDEN list items | jq ".[] | select( .name == \"$ITEM\" ).id" | tr -d '"'`
`
If "ITEM_ID" is empty, I invoke command to create it.

In the past, the script worked with the official BitWarden cloud account. It doesn't work with my own vaultwarden server.

Regard.
Flo

**Reply by BlackDex** [reply]
There should be no difference in the workings.
There is no reference or linkage between the piping and server, not sure where you get that from.

What i mean is, instead of running `echo ... | bw encode | x y z` just run `bw edit item ...`.
Try to use the `--raw` option which  might show some more detailed info maybe?

**Reply by Flo-Apitech** [reply]
Hi,

I found my problem with your last message. 

In my script, I have this var : ITEM_JSON
This var is Json format and contain  with fields "collectionIds" and "organizationId". When migrating on my own server, system generated new ID and I forgot change this value in my script...

Also, when you use `bw edit item`, that command don't use "collectionIds" and "organizationId" but only field "name".

Thank you very much for your help.
